test/utils/strings.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Compares two strings and returns whether they are the same.
+ * @param a - The first string to compare.
+ * @param b - The second string to compare.
+ * @param ignoreCase - Optional. Specifies whether the comparison should be case-insensitive. Default is true.
+ * @returns True if the strings are the same, false otherwise.
+ */
+const sameString = (a: string, b: string, ignoreCase = true): boolean => {
+    return ignoreCase ? a.toLowerCase() === b.toLowerCase() : a === b;
+};
+
+/**
+ * Aligns a hexadecimal string to a specified number of bytes by adding leading zeros if necessary.
+ *
+ * @param hex - The hexadecimal string to align.
+ * @param bytes - The desired number of bytes.
+ * @returns The aligned hexadecimal string.
+ * @throws Error if the aligned string exceeds the specified number of bytes.
+ */
+const alignHexString = (hex: string, bytes: number): string => {
```

Comment:
nit: Ethers provides this already: https://docs.ethers.org/v6/api/utils/#zeroPadBytes and https://docs.ethers.org/v6/api/utils/#zeroPadValue depending on left or right padding is required.